vertx-service-discovery/src/test/java/io/vertx/servicediscovery/impl/DiscoveryImplClusteredTest.java

```diff
@@ -19,21 +19,26 @@
 import io.vertx.core.Vertx;
 import io.vertx.core.VertxOptions;
 import io.vertx.servicediscovery.ServiceDiscoveryOptions;
+import io.vertx.test.fakecluster.FakeClusterManager;
 import org.junit.Before;
 
 import static com.jayway.awaitility.Awaitility.await;
 
 /**
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
  */
-public class DiscoveryImplClusteredTest extends DiscoveryImplTest {
+public class DiscoveryImplClusteredTest extends DiscoveryImplTestBase {
 
   @Before
   public void setUp() {
-    Vertx.clusteredVertx(new VertxOptions().setClusterHost("127.0.0.1"), ar -> {
+    FakeClusterManager.reset();
+    VertxOptions options = new VertxOptions()
+      .setClusterManager(new FakeClusterManager())
+      .setClusterHost("127.0.0.1");
+    Vertx.clusteredVertx(options, ar -> {
```

Comment:
I'm a bit worried we don't test the same thing anymore. Typically, with a "Real" cluster manager we have latency in the replication, here it's instantaneous. Of course, it should not matter, but in practice, it tends to matter.

The existing approach was far from perfect, and on a real cluster, the latency and replication error can be terrible.

Ideally, we would need a _wiremock_ for cluster managers where we can adapt the condition and response to be sure the code still behave correctly.